src/Features/Core/Portable/Shared/Options/ServiceFeatureOnOffOptions.cs

```diff
@@ -8,11 +8,33 @@ internal static class ServiceFeatureOnOffOptions
     {
         public const string OptionName = "ServiceFeaturesOnOff";
 
+        private const bool CSharpClosedFileDiagnosticsEnabledByDefault = false;
+        private const bool DefaultClosedFileDiagnosticsEnabledByDefault = true;
+
         /// <summary>
         /// this option is solely for performance. don't confused by option name. 
         /// this option doesn't mean we will show all diagnostics that belong to opened files when turned off,
         /// rather it means we will only show diagnostics that are cheap to calculate for small scope such as opened files.
         /// </summary>
-        public static readonly PerLanguageOption<bool> ClosedFileDiagnostic = new PerLanguageOption<bool>(OptionName, "Closed File Diagnostic", defaultValue: true);
+        public static readonly PerLanguageOption<int> ClosedFileDiagnostic = new PerLanguageOption<int>(OptionName, "Closed File Diagnostic", defaultValue: -1);
```

Comment:
We have identified that having a boolean state doesn't suffice for this option anymore. Reasons:

1. We want to track explicitly when the check box setting is untouched by the user versus turned off by unchecking - it is very likely we will turn this option back on by default for some scope of background analysis, and we don't want to break the users who explicitly turned it off, but also want the users who didn't ever notice this option to get back the background analysis.
2. We are considering adding more states for background analysis - e.g. run for current documents, open documents, current project, entire solution (just like the error list). We will need more state values for this option.

To summarize, this is a true 3-state option now, likely to become a 3+ multistate option in future. This change deprecates the 2-state option for the newer version, which needs the integral values to be emitted into the settings.